titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/pod/v1/V1SpecPodFactory.java

```diff
@@ -272,6 +277,30 @@ private V1Container buildV1ExtraContainers(BasicContainer extraContainer) {
                 .imagePullPolicy(DEFAULT_IMAGE_PULL_POLICY);
     }
 
+
+    private List<V1Volume> buildV1Volumes(List<Volume> volumes) {
+        if (volumes == null) { return null;}
+        return volumes.stream().map(this::buildV1Volume).collect(Collectors.toList());
+    }
+
+    private V1Volume buildV1Volume(Volume volume) {
+        V1FlexVolumeSource flexVolume = getV1FlexVolume(volume);
+        return new V1Volume()
+                .name(volume.getName())
+                .flexVolume(flexVolume);
+    }
+
+    private V1FlexVolumeSource getV1FlexVolume(Volume volume) {
+        SharedContainerVolumeSource sharedContainerVolumeSource = volume.getSharedContainerVolumeSource();
+        if (sharedContainerVolumeSource == null) { return null; }
```

Comment:
Do you know what happens if this returns null and it is set to null higher up?

Reply:
Yes. On these Volume types, there a ton of BLASource fields. One for every kind of volume you can think of. 99% of them are null.

When these fields are null, all the storage drivers just know not to worry about them.